drivers/video/video_shell.c

```diff
@@ -0,0 +1,618 @@
+/*
+ * Copyright (c) 2024 tinyVision.ai Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/video.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/shell/shell.h>
+
+#define FPS(frmival) DIV_ROUND_CLOSEST((frmival)->denominator, (frmival)->numerator)
+
+static struct video_buffer *vbuf;
+
+static int video_shell_check_device(const struct shell *sh, const struct device *dev)
+{
+	if (dev == NULL) {
+		shell_error(sh, "could not find a video device with that name");
+		return -ENODEV;
+	}
+
+	if (!DEVICE_API_IS(video, dev)) {
```

Comment:
Yes thank you, this would allow the `SW_GENERATOR` device to work.
![scrot_20241202_185003_434x70](https://github.com/user-attachments/assets/2f093366-adff-4fed-ae9e-cddc229a0dab)